pkg/gce-pd-csi-driver/controller.go

```diff
@@ -1327,3 +1351,95 @@ func pickRandAndConsecutive(slice []string, n int) ([]string, error) {
 	}
 	return ret, nil
 }
+
+func (gceCS GCEControllerServer) CheckAndUpdateLastKnownInstanceOps(ctx context.Context, project, location, instanceName string) error {
+	ops := gceCS.opsCache.InstanceOps.GetOps(common.CreateInstanceKey(project, location, instanceName))
+	if ops == nil {
+		return nil
+	}
+
+	for _, op := range ops {
+		done, err := gceCS.CloudProvider.CheckOpDoneStatus(ctx, project, location, op.Name)
+		if err != nil {
+			return err
+		}
+
+		if !done {
```

Comment:
This check forces new Op to be added only after all Ops on an Instance is done, which means there can be only 1 Op running on an Instance at any given time. what is the reason to have an additional disk-instance map?